spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/AppRegistry.java

```diff
@@ -81,8 +83,22 @@ public AppRegistration save(String name, ApplicationType type, URI uri) {
 		return apps;
 	}
 
+	/**
+	 * Deletes an {@link AppRegistration}. If the {@link AppRegistration} does not
+	 * exist, a {@link NoSuchAppRegistrationException} will be thrown.
+	 *
+	 * @param name Name of the AppRegistration to delete
+	 * @param type Type of the AppRegistration to delete
+	 */
 	public void delete(String name, ApplicationType type) {
-		this.uriRegistry.unregister(key(name, type));
+		final AppRegistration appRegistrationToDelete = this.find(name, type);
+
+		if (appRegistrationToDelete != null) {
```

Comment:
Given `appRegistrationToDelete` is used only in this line, can we move `find` here directly and perform null check on return value?